### PR TITLE
Fast-Integrationtests: adjust pipeline execution for skr-test

### PR DESCRIPTION
**Pull request by tillknuesting**
Currently, some prow jobs are run that are unrelated to the skr-test that is changed.

Make `pre-main-kyma-integration-k3d-central-app-connectivity-compass` only to be run if manually triggered with `always_run: false`
` pre-main-kyma-skr-eventing 
` only to be run on eventing-test folder

### CODE REVIEW

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 1]
```diff
@@ -321,7 +321,7 @@ presubmits: # runs on PRs
         preset-kyma-integration-compass-enabled: "true"
         preset-kyma-integration-kcp-client-credentials: "true"
         preset-kyma-integration-subaccounnt: "true"
-      run_if_changed: '^((tests/fast-integration\S+|resources/eventing\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
+      run_if_changed: '^((resources/eventing\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
```
Comment (akgalwas) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:324]: I'm not sure about this change. That's true we don't want to run `pre-main-kyma-skr-eventing` each time someone changed anything in Fast Integration Tests but we do want to quickly find regressions in the eventing test. I would rather modify the regexp to execute the pipeline in case of change in `tests/fast-integration)/eventing-test/` folder.

Comment (akgalwas) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:324]: Please fix tests to make `pre-main-test-infra-validate-prow-jobs` running.

File: prow/jobs/kyma/kyma-integration.yaml [hunk 2]
```diff
@@ -256,7 +256,7 @@ presubmits: # runs on PRs
         preset-kyma-integration-compass-dev: "true"
         preset-kyma-integration-compass-enabled: "true"
         preset-sa-vm-kyma-integration: "true"
-      run_if_changed: '^((tests/fast-integration\S+|resources\S+|installation\S+|tools/kyma-installer\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
+      run_if_changed: '^((resources\S+|installation\S+|tools/kyma-installer\S+)(\.[^.][^.][^.]+$|\.[^.][^dD]$|\.[^mM][^.]$|\.[^.]$|/[^.]+$))'
```
Comment (akgalwas) [inline prow/jobs/kyma/kyma-integration.yaml:259]: The same here. I would rather modify the regexp to execute the pipeline in case of change in tests/fast-integration)/compass-test/ folder.

File: prow/jobs/kyma/kyma-integration-gardener.yaml (hunk 1 above)
Comment (tillknuesting) [reply prow/jobs/kyma/kyma-integration-gardener.yaml:324]: can you please elaborate on this?

**Comment by tillknuesting** [comment]
@akgalwas could you have a look again?

**Comment by tillknuesting** [comment]
@Ressetkk could please review it?

**Comment by Ressetkk** [comment]
@tillknuesting if possible, please fix the unit tests if the use case got changed. Right now the tests assume this job has `run_if_changed` defined, thus the job fails.

**Comment by tillknuesting** [comment]
@Ressetkk, could I kindly ask for a review again?

**Review by ralikio — APPROVED** [review]

File: development/tools/jobs/kyma/kyma_integration_test.go [hunk 3]
```diff
@@ -35,23 +35,6 @@ func TestKymaIntegrationJobsPresubmit(t *testing.T) {
...
-			expRunIfChangedRegex: "^((tests/fast-integration\\S+|resources\\S+|installation\\S+|tools/kyma-installer\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))",
-			expRunIfChangedPaths: []string{
-				"resources/values.yaml",
-				"installation/file.yaml",
-			},
-			expNotRunIfChangedPaths: []string{
-				"installation/README.md",
-				"installation/test/test/README.MD",
-			},
-		},
```
Comment (Ressetkk) [inline development/tools/jobs/kyma/kyma_integration_test.go:54]: I would rather avoid removing the current test cases as we don't have clear decision on how to move on with old job tests. Can you justify why did you remove this test case?

Comment (tillknuesting) [reply development/tools/jobs/kyma/kyma_integration_test.go]: I added back the unit test but reduced the scope of the test cases.

**Comment by tillknuesting** [comment]
@Ressetkk, could I kindly ask for a review again?

**Review by Ressetkk — APPROVED** [review]
